app/helpers/argo_helper.rb

```diff
@@ -147,7 +147,7 @@ def render_buttons(doc, object = nil)
       # date=embargo_data.split.last
       if text != 'released'
         # TODO: add a date picker and button to change the embargo date for those who should be able to.
-        buttons << {:label => 'Update embargo', :url => embargo_form_item_path(pid)} if current_user.is_admin
+        buttons << {:label => 'Update embargo', :url => embargo_form_item_path(pid)} if can?(:manage_embargo, object)
```

Comment:
Assuming that `can?(:manage_embargo, object)` includes `User#is_admin` as it's highest precedence, this will behave the same way and may be preferred, but if it falls into requesting object permission it could behave differently.  It may be that this should not change and only Argo admins can do this, regardless of object permissions.